presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -307,12 +313,25 @@ public void close()
         private final Map<LifespanAndStage, LifespanAndStageState> committedRecoverableLifespanAndStages = new HashMap<>();
 
         private final LifespanCommitter lifespanCommitter;
+        private final List<ListenableFuture<?>> commitFutures = new ArrayList<>();
 
         LifespanAndStageStateTracker(LifespanCommitter lifespanCommitter)
         {
             this.lifespanCommitter = requireNonNull(lifespanCommitter, "lifespanCommitter is null");
         }
 
+        public void waitForAllLifespanCommitted()
+        {
+            ListenableFuture<?> future = whenAllSucceed(commitFutures).call(() -> null, directExecutor());
+            try {
+                getFutureValue(future, PrestoException.class);
```

Comment:
@shixuan-fan Sorry, one comment

I would avoid using the `getFutureValue` as it is

1. Makes the `Interrupted` and cancellation semantics unclear
2. Flattens the `ExecutionException`. We don't want to flatten the `ExecutionException`, unles it is `PrestoException`, as we will miss the `waitForAllLifespanCommitted` point in the stacktrace.

Reply:
Done. You might want to take a second look just to see if it addresses your comment. And never feel sorry to leave comment to improve code quality :D

Reply:
Thanks for addressing ;-)